runtime/collect/StackMapParser.cpp

```diff
@@ -39,26 +39,35 @@ void parseStackMap() {
       void *ip = (void *)(func.FunctionAddress + InstructionOffset);
       int32_t NumDeopts = *(int32_t *)(stackMapRecord + 24 + 2 * 12);
       uint16_t RelocationOffset = 0;
-      for (uint16_t j = 4 + NumDeopts; j < NumLocations; j += 2) {
-        uint8_t type = *(uint8_t *)(stackMapRecord + 16 + j * 12);
-        if (type == 5) {
+      for (uint16_t j = 3 + NumDeopts; j < NumLocations; j += 2) {
```

Comment:
What part of the schema is this looking at, and what's the difference between offset 4 and offset 3? Some further comments tying this parser to [the schema](https://llvm.org/docs/StackMaps.html#stack-map-format) would be nice - I have no doubt it's correct, but it's difficult to follow the byte arithmetic to make sure it's correct.